lark/tree_templates.py

```diff
@@ -9,14 +9,18 @@
 from lark import Tree, Transformer
 from lark.exceptions import MissingVariableError
 
+Branch = Union[Tree[str], str]
 TreeOrCode = Union[Tree[str], str]
+MatchResult = Dict[str, Tree]
 _TEMPLATE_MARKER = '$'
 
 
 class TemplateConf:
     """Template Configuration
 
     Allows customization for different uses of Template
+
+    parse() must return a Tree instance.
     """
 
     def __init__(self, parse=None):

```

Comment:
This could be annotated with `Callable[[str], Tree]`

Reply:
I agree, but there's a lot of "could be" to add there. We can do that in a different PR.